pkg/services/ngalert/schedule/schedule.go

```diff
@@ -202,11 +162,6 @@ func (sch *schedule) DeleteAlertRule(key ngmodels.AlertRuleKey) {
 	}
 	// stop rule evaluation
 	ruleInfo.stop()
-
-	// Our best bet at this point is that we update the metrics with what we hope to schedule in the next tick.
-	alertRules := sch.schedulableAlertRules.all()
-	sch.metrics.SchedulableAlertRules.Set(float64(len(alertRules)))
-	sch.metrics.SchedulableAlertRulesHash.Set(float64(hashUIDs(alertRules)))
```

Comment:
This is moved to the newly created store. Yes, it will give it a lag of up to 10 seconds but it will be consistent in HA mode.